### ISSUE TITLE
Gradle won't install

### ISSUE THREAD

**Issue by rodrigomalara** [sdkman/sdkman-cli#445]
I'm a newbie and just installed sdk on a Windows 8.1 machine running cygwing.
Installation went smooth but when I try to install gradle I get the following message:

Rodrigo@rodrigo-pc ~
$ sdk install gradle

Downloading: gradle 2.14.1

In progress...

######################################################################## 100.0%
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /home/Rodrigo/.sdkman/archives/gradle-2.14.1.zip or
        /home/Rodrigo/.sdkman/archives/gradle-2.14.1.zip.zip, and cannot find /home/Rodrigo/.sdkman/archives/gradle-2.14.1.zip.ZIP, period.

Stop! The archive was corrupt and has been removed! Please try installing again.

Thanks in advance for any help

**Comment by xasx** [comment]
I experience this as well. Apparently `curl` is not following the redirect there:

```
$ curl --progress-bar --location 'https://api.sdkman.io/download/gradle/2.14.1?platform=Linux' 
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>301 Moved Permanently</title>
</head><body>
<h1>Moved Permanently</h1>
<p>The document has moved <a href="https://downloads.gradle.org/distributions/gradle-2.14.1-bin.zip">here</a>.</p>
<hr>
<address>Apache/2.2.3 (CentOS) Server at services.gradle.org Port 443</address>
</body></html>
```

**Comment by xasx** [comment]
Might be an issue with some transparent proxies on my work network. However, updating sdkman itself worked.

So does installing vertx.

**Comment by xasx** [comment]
Comparing my own outputs and matching curl's output to wget's, I think that it is a problem on Gradle's side. Something with the 301 redirect probably.

**Comment by rodrigomalara** [comment]
Thanks xasx for the help.

I've installed sdkman today but anyway forced an upgrade.
Still not working.

I tried installing vertx and it works.
Is there a way to tell sdkman where it should point for gradle distribution ?

**Comment by xasx** [comment]
@rodrigomalara sdkman retrieves downloads from a resource on `sdkman.io`, e.g.

https://api.sdkman.io/download/gradle/2.14.1?platform=Linux 
(your browser will download this just fine)

It is implicitly being built from the pattern

`https://api.sdkman.io/download/<candidate>/<version>?platform=<platform>`

I haven't found a way to override this for specific candidates so far. Probably it would be enough to redirect to the _downloads.gradle.org_ URI instead of the services.gradle.org one. 
However, from a Gradle distributor point of view, I don't know whether it is desirable.

**Comment by xasx** [comment]
Workaround: Download manually and copy the `-bin.zip` over to `$HOME/.sdkman/archives/gradle-$VERSION.zip` (without the bin), then install.

**Comment by xasx** [comment]
Created a topic in their forum to help sorting this out:

https://discuss.gradle.org/t/zip-download-redirect-and-sdkman/18629?u=xasx

**Comment by marc0der** [comment]
I've just tried this locally and have no problem whatsoever. Curl follows the redirects without any issues. Could you please try again and let me know if this is resolved?

@xasx the download endpoint that you were calling will give a 404 because the HEAD method is not supported, only GET.

**Comment by xasx** [comment]
@marc0der In fact it works now, at least for me.

@rodrigomalara Perhaps you can also re-check.

**Comment by marc0der** [comment]
No further response so closing this issue.

**Comment by tbodine88** [comment]
I experienced this same issue last night, 

I finally resorted to downloading gradle, unzipping and installing manually. Was closing premature?

BTW I thought my issue might be caused by installing unxutils.zip instead of cygwin, but when I removed zip and unzip in unxutils I still got the same "Stop! The archive was corrupt and has been removed!" error

**Comment by pchartrand** [comment]
same problem using  curl -s get.sdkman.io

inspecting the install script I found the url called for downloading is 

https://api.sdkman.io/1/res?platform=Linux&purpose=install

this returns a 404 from a tomcat server on bintray.com

https://bintray.com/artifact/download/sdkman/generic/sdkman-cli-5.2.0%2B125.zip

**Comment by marc0der** [comment]
This had nothing to do with Gradle not downloading. What you were experiencing was due to the sdkman binary not being published to bintray after the release of 5.2.0. I rectified this.

**Comment by pchartrand** [comment]
Thanks for correcting the situation.
